src/p2p/net_node.inl

```diff
@@ -83,6 +83,19 @@ static inline boost::asio::ip::address_v4 make_address_v4_from_v6(const boost::a
   return boost::asio::ip::address_v4(v4);
 }
 
+static inline uint16_t get_port(const epee::net_utils::network_address &na)
```

Comment:
Looks like this should be a function of `network_address`.

* Add `virtual std::uint16_t port() const = 0;` [here](https://github.com/monero-project/monero/blob/master/contrib/epee/include/net/net_utils_base.h#L232) to declare the interface.
* Add `virtual std::uint16_t port() const override { return value.port(); }` [here](https://github.com/monero-project/monero/blob/master/contrib/epee/include/net/net_utils_base.h#L262) to bind every instance of `network_address` to the interface.
* Add `std::uint16_t port() const { return self ? self->port() : 0 }` [here](https://github.com/monero-project/monero/blob/master/contrib/epee/include/net/net_utils_base.h#L308) to handle the wrapper (which handles the case where the `network_address` was never set).

Then just call `na.port()` everywhere.